### PR TITLE
Add table oci_core_public_ip_pool. Closes #180

### LINKED ISSUE #180: Add table oci_core_public_ip_pool
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by rajeshbal65**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Test names: [ 'tests/oci_core_public_ip_pool' ]
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_core_public_ip_pool []

PRETEST: tests/oci_core_public_ip_pool

TEST: tests/oci_core_public_ip_pool
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # oci_core_public_ip_pool.named_test_resource will be created
  + resource "oci_core_public_ip_pool" "named_test_resource" {
      + cidr_blocks    = (known after apply)
      + compartment_id = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
      + defined_tags   = (known after apply)
      + display_name   = "steampipetest974"
      + freeform_tags  = {
          + "Name" = "steampipetest974"
        }
      + id             = (known after apply)
      + state          = (known after apply)
      + time_created   = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + freeform_tags   = {
      + "Name" = "steampipetest974"
    }
  + lifecycle_state = (known after apply)
  + region          = "ap-mumbai-1"
  + resource_id     = (known after apply)
  + resource_name   = "steampipetest974"
  + tenancy_ocid    = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
oci_core_public_ip_pool.named_test_resource: Creating...
oci_core_public_ip_pool.named_test_resource: Creation complete after 2s [id=ocid1.publicippool.oc1.ap-mumbai-1.amaaaaaa6igdexaac2igqw4nbjvurn5xkux4srrxyjii6lmxfr4olnaxhk6q]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

freeform_tags = tomap({
  "Name" = "steampipetest974"
})
lifecycle_state = "ACTIVE"
region = "ap-mumbai-1"
resource_id = "ocid1.publicippool.oc1.ap-mumbai-1.amaaaaaa6igdexaac2igqw4nbjvurn5xkux4srrxyjii6lmxfr4olnaxhk6q"
resource_name = "steampipetest974"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"

Running SQL query: test-get-query.sql

[
  {
    "display_name": "steampipetest974",
    "freeform_tags": {
      "Name": "steampipetest974"
    },
    "id": "ocid1.publicippool.oc1.ap-mumbai-1.amaaaaaa6igdexaac2igqw4nbjvurn5xkux4srrxyjii6lmxfr4olnaxhk6q",
    "lifecycle_state": "ACTIVE"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "display_name": "steampipetest974",
    "freeform_tags": {
      "Name": "steampipetest974"
    },
    "id": "ocid1.publicippool.oc1.ap-mumbai-1.amaaaaaa6igdexaac2igqw4nbjvurn5xkux4srrxyjii6lmxfr4olnaxhk6q",
    "lifecycle_state": "ACTIVE"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "region": "ap-mumbai-1",
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "steampipetest974"
  }
]
✔ PASSED

POSTTEST: tests/oci_core_public_ip_pool

TEARDOWN: tests/oci_core_public_ip_pool

SUMMARY:

1/1 passed.cd 
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
+--------------+-------------------------------------------------------------------------------------------------+--------+---------------------+-------------+
| display_name | id                                                                                              | state  | time_created        | region      |
+--------------+-------------------------------------------------------------------------------------------------+--------+---------------------+-------------+
| test         | ocid1.publicippool.oc1.ap-mumbai-1.amaaaaaa6igdexaasjtzyk4xr7ia4ce4pvflt5hdtvjexjfczktq464cs2ea | ACTIVE | 2021-07-28 10:56:25 | ap-mumbai-1 |
| test         | ocid1.publicippool.oc1.ap-mumbai-1.amaaaaaa6igdexaa44zojv4aee3xqpg5uvsxp7ufhxi7gyon7q2nermts6aa | ACTIVE | 2021-07-28 09:36:11 | ap-mumbai-1 |
+--------------+-------------------------------------------------------------------------------------------------+--------+---------------------+-------------+
```

### List public IP pool which are not active

```sql
select
  display_name,
  id,
  lifecycle_state as state
from
  oci_core_public_ip_pool
where
  lifecycle_state <> 'ACTIVE';

+--------------+----+-------+
| display_name | id | state |
+--------------+----+-------+
+--------------+----+-------+
```
</details>

### CODE REVIEW

File: docs/tables/oci_core_public_ip_pool.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Table: oci_core_public_ip
+
+A public IP pool is a set of public IP addresses represented as one or more IPv4 CIDR blocks. Resources like load balancers and compute instances can be allocated public IP addresses from a public IP pool.
+
+## Examples
+
```
Comment (bigdatasourav) [inline docs/tables/oci_core_public_ip_pool.md:6]: can we add more examples

File: oci/table_oci_core_public_ip_pool.go [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+		GetMatrixItem: BuildCompartementRegionList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "display_name",
+				Description: "A user-friendly name. Does not have to be unique.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "The OCID of the public IP pool..",
```
Comment (bigdatasourav) [inline oci/table_oci_core_public_ip_pool.go:37]:
```suggestion
				Description: "The OCID of the public IP pool.",
```

File: oci/table_oci_core_public_ip_pool.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "time_created",
+				Description: "The date and time the public IP pool was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
+			},
+			{
+				Name:        "cidr_blocks",
```
Comment (bigdatasourav) [inline oci/table_oci_core_public_ip_pool.go:53]: json columns will be in last

File: oci/table_oci_core_public_ip_pool.go [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+				Description: "The public IP pool's current state.",
```
Comment (bigdatasourav) [inline oci/table_oci_core_public_ip_pool.go:54]: description is not correct

File: oci/table_oci_core_public_ip_pool.go [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	plugin.Logger(ctx).Error("getCorePublicIPPool", "Compartment", compartment, "OCI_REGION", region)
+
+	// Restrict the api call to only root compartment/ per region
+	if !strings.HasPrefix(compartment, "ocid1.tenancy.oc1") {
+		return nil, nil
+	}
+	id := d.KeyColumnQuals["id"].GetStringValue()
+
+	// handle empty snapshot id in get call
```
Comment (bigdatasourav) [inline oci/table_oci_core_public_ip_pool.go:161]:
```suggestion
	// handle empty public ip pool id in get call
```

**Review by bigdatasourav — CHANGES_REQUESTED** [review]
@rajeshbal65 some suggested changes.

File: oci-test/tests/oci_core_public_ip_pool/variables.tf [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+output "lifecycle_state" {
+  value = oci_core_public_ip_pool.named_test_resource.state
+}
+
+output "freeform_tags" {
+  value = oci_core_public_ip_pool.named_test_resource.freeform_tags
+}
+
```
Comment (bigdatasourav) [inline oci-test/tests/oci_core_public_ip_pool/variables.tf:65]:
```suggestion
```

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**